src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -37,30 +41,27 @@ public WorkItem(SendOrPostCallback callback, object? data, ManualResetEventSlim?
             }
         }
 
-        private static JSSynchronizationContext? MainThreadSynchronizationContext;
-        private readonly QueueType Queue;
-        private readonly Action _DataIsAvailable;// don't allocate Action on each call to UnsafeOnCompleted
-
-        private JSSynchronizationContext()
+        private JSSynchronizationContext(Thread targetThread, IntPtr targetThreadId)
             : this(
-                Thread.CurrentThread,
+                targetThread, targetThreadId,
                 Channel.CreateUnbounded<WorkItem>(
                     new UnboundedChannelOptions { SingleWriter = false, SingleReader = true, AllowSynchronousContinuations = true }
                 )
             )
         {
         }
 
-        private JSSynchronizationContext(Thread mainThread, QueueType queue)
+        private JSSynchronizationContext(Thread targetThread, IntPtr targetThreadId, WorkItemQueueType queue)
```

Comment:
We need a comment somewhere in this .cs file explaining the model behind why there are thread-local current queues and the queues have target thread IDs, how it's meant to work etc

Reply:
Agreed, I added FIXME comment for now, because we are still figuring this out

Reply:
Added link to `threads.md` instead. And eventually we will explain things there.